### ISSUE TITLE
Turn insert_final_newline on by default

### ISSUE THREAD

**Issue by egor-n** [shyiko/ktlint#103]
What do you think about introducing the "file should end with a newline" rule?

[Here's the Checkstyle rule](http://checkstyle.sourceforge.net/config_misc.html#NewlineAtEndOfFile) for this.

**Comment by shyiko** [comment]
Hi @egor-n 

You can enable it by adding this to your [.editorconfig](https://github.com/shyiko/ktlint#editorconfig)
```ini
[*.{kt,kts}]
insert_final_newline=true
```

**Comment by egor-n** [comment]
I should have thought of that. Thank you!

**Comment by shyiko** [comment]
No worries. Documentation wasn't very clear. If you don't mind I'll keep this ticket open until insert_final_newline is set to true by default.

**Comment by vanniktech** [comment]
Right now it would not fail if there are two new lines at the end of a file. I guess that could be optimized, right?

**Comment by shyiko** [comment]
@vanniktech 2 or more blank lines should trigger "Needless blank line(s)" error. If you don't please submit a bug report.

**Comment by vanniktech** [comment]
There you go #108

**Comment by shyiko** [comment]
Instead of enabling this option by default I went with improving ktlint's [.editorconfig](http://editorconfig.org/) support by automatically falling back to "[*]" for property resolution. This way people who have .editoconfig with `insert_final_newline = true` under `[*]` won't have to do anything else to enable it and others won't be surprised by the change. 

"not enabled unless .editorconfig says otherwise" behavior is currently due to the fact that neither Jetbrains's nor Google's styleguide says anything about newline at the end of file. I'll try to clarify it before making insert_final_newline on by default. 

`insert_final_newline = true` under `[*]` support included in [ktlint@0.13.0](https://github.com/shyiko/ktlint/releases/tag/0.13.0).

**Comment by thauk-copperleaf** [comment]
Would love to see this enabled by default, especially because git diff warns about this. (A lot of tooling warns about it, actually!)
